### DISCUSSION TITLE
Base Stations no longer reachable

### DISCUSSION THREAD

**Discussion by MilesAheadToo** [ESPresense/ESPresense#549] [Support]
I had ESPresense working well with HA to detect arrival home in order to open the gate and disable the alarm if necessary. I want to extend the HA automation to include departure and automatic setting of the alarm if nobody is at home.

However, the base stations are no longer detecting the beacons. I can run other apps to detect BLE beacons, such as BLE Scanner on my Android phone, and the beacons are all visible. This leads me to believe the issue is not with the beacons.

I suspect the issue arises from the MQTT configuration I am using. I greatly appreciate any pointers.

This is a diagnostic from one of the base stations.
[mqtt-893b5dcc7bd3c9150db49010c49d0ed6-ESPresense Gate-5996b97c6d0c33545da67453b55aa4b5.json.txt](https://github.com/ESPresense/ESPresense/files/9121919/mqtt-893b5dcc7bd3c9150db49010c49d0ed6-ESPresense.Gate-5996b97c6d0c33545da67453b55aa4b5.json.txt)

This is a partial screenshot of the HA UI panel that I am using to bring all the ESPresense info together. As is clear from the screenshot, no beacons are detected by any of the 3 base stations because `Count=0` for each base station, the beacon states are all `not_home` and the distances are all `unknown` though all the beacons are at home.

![image](https://user-images.githubusercontent.com/78782836/179252316-38e843c6-a90c-40f6-af64-0222b0f3239a.png)

I use `mqtt: !include mqtt.yaml` in my configuration.yaml file and the following mqtt.yaml file.
[mqtt.txt](https://github.com/ESPresense/ESPresense/files/9122110/mqtt.txt)

Also, when I try to connect to the base stations through the HA UI I get an error when I click on the `Visit Device` link in the HA UI.

![image](https://user-images.githubusercontent.com/78782836/179257744-d18ada3c-71ce-418d-b7a6-39f8e6ef6887.png)

This is the error.

![image](https://user-images.githubusercontent.com/78782836/179258125-f56cd483-4336-43a4-b596-10a8b7bebd90.png)

**Comment by MilesAheadToo** [comment]
In case it is useful, this is the MQTT diagnostic.
[config_entry-mqtt-893b5dcc7bd3c9150db49010c49d0ed6.json.txt](https://github.com/ESPresense/ESPresense/files/9122591/config_entry-mqtt-893b5dcc7bd3c9150db49010c49d0ed6.json.txt)

**Comment by MilesAheadToo** [comment]
And this is the log file

[mqtt_log.txt](https://github.com/ESPresense/ESPresense/files/9122973/mqtt_log.txt)

**Comment by DTTerastar** [comment]
If you can't connect to it via Wifi, then the node can't connect to MQTT.  Did anything change with your Wifi?  The nodes also didn't auto update, are they in a separate IOT network?  Have you tried power cycling them?

**Reply by MilesAheadToo** [reply]
Sorry for the slow reply @DTTerastar.

It appears that the M5 base stations go to sleep every so often. I did a manual reboot of each and got them connected to the WiFi again. Then I noticed that after about 24 hours they were reporting incorrect values. I could still access them, so I used the HA UI to restart them, and the values reported were correct again.

So I am thinking of adding an automation that restarts the M5 base stations every day. I just have to be sure that they do not trigger a gate opening after the restart.

**Reply by DTTerastar** [reply]
3.0b1 has a fix for this

**Reply by MilesAheadToo** [reply]
Is this update available via the usual HA update process.
